app/assets/javascripts/components/samples.jsx

```diff
@@ -74,15 +85,37 @@ class Samples extends React.Component {
   applyClass(status) {
     if(status === 'COMPLETE') {
       return 'complete';
-    } else if (status === 'UPLOADING' || status === 'IN PROGRESS') {
+    } else if (status === 'WAITING') {
+      return 'waiting';
+    } else if (status === 'INPROGRESS') {
       return 'uploading';
-    } else if (status === 'INITIALIZING') {
-      return 'initializing';
-    } else {
+    } else if (status === 'FAILED') {
       return 'failed';
     }
   }
 
+  applyChunkStatusClass(postProcess, alignment, hostFiltering) {
+    if (postProcess) {
+      return postProcess === 'LOADED' ? 'complete' : 'uploading';
+    } else if(alignment) {
+      return alignment === 'FAILED' ? 'failed' : 'uploading';
+    } else if(hostFiltering) {
+      return hostFiltering === 'FAILED' ? 'failed' : 'uploading';
+    }
+  }
+
+  getChunckedStatus(postProcess, alignment, hostFiltering) {
+    if (postProcess && alignment === 'LOADED' && hostFiltering === 'LOADED') {
+      return  'Post Processing';
+    } else if (postProcess === null && alignment && hostFiltering === 'LOADED') {
+      return 'Alignment';
+    } else if (postProcess === null && alignment === null && hostFiltering) {
+      return 'Host Filtering';
+    } else {
+      return 'WAITING';
```

Comment:
where is FAILED status?

Reply:
this method only shows what stages are presently running for that sample, not the status.
the name will be changed to getChunkedStage @yunfangjuan